src/Tgstation.Server.Host/Components/InstanceManager.cs

```diff
@@ -547,7 +547,7 @@ private void CheckSystemCompatibility()
 
 			// This runs before the real socket is opened, ensures we don't perform reattaches unless we're fairly certain the bind won't fail
 			// If it does fail, DD will be killed.
-			SocketExtensions.BindTest(serverPortProvider.HttpApiPort, true);
+			SocketExtensions.BindTest(Convert.ToUInt16(serverAddressProvider.AddressEndPoint.Port), true);
```

Comment:
is this safe? IPEndPoint uses an int to store the port, not a ushort

Reply:
Yeah, it's just framework devs being lazy, backwards compat, or catering to lowest common denominator